strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/proxied/LocalStorageProxyRepositoryExpiredArtifactsCleaner.java

```diff
@@ -52,18 +58,52 @@ public void cleanup(final Integer lastAccessedTimeInDays,
 
         final List<ArtifactEntry> artifactEntries = artifactEntryService.findMatching(searchCriteria,
                                                                                       PagingCriteria.ALL);
+        filterAccessibleProxiedArtifacts(artifactEntries);
 
         if (CollectionUtils.isEmpty(artifactEntries))
         {
             return;
         }
 
         logger.debug("Cleaning artifacts {}", artifactEntries);
-
         deleteFromDatabase(artifactEntries);
         deleteFromStorage(artifactEntries);
     }
 
+    private void filterAccessibleProxiedArtifacts(final List<ArtifactEntry> artifactEntries)
+    {
+        if (CollectionUtils.isEmpty(artifactEntries))
+        {
+            return;
+        }
+        for (final Iterator<ArtifactEntry> it = artifactEntries.iterator(); it.hasNext(); )
+        {
+            final ArtifactEntry artifactEntry = it.next();
+            final Storage storage = configurationManager.getConfiguration().getStorage(artifactEntry.getStorageId());
+            final Repository repository = storage.getRepository(artifactEntry.getRepositoryId());
+            if (!repository.isProxyRepository())
+            {
+                logger.debug("Repository {} is not a proxy repository.", repository.getId());
```

Comment:
I think will be causing a lot of unnecessary debugging output and we could probably safely remove it.